directional_pool.go

```diff
@@ -0,0 +1,235 @@
+package udnssdk
+
+import (
+	"fmt"
+	"log"
+	"time"
+)
+
+// DirectionalPoolsService manages 'account level' 'geo' and 'ip' groups for directional-pools
+type DirectionalPoolsService struct {
+	client *Client
+}
+
+// AccountLevelGeoDirectionalGroupDTO wraps an account-level, geo directonal-group response
+type AccountLevelGeoDirectionalGroupDTO struct {
+	Name        string   `json:"name"`
+	Description string   `json:"description"`
+	Codes       []string `json:"codes"`
+}
+// IPAddrDTO wraps an IP address range or CIDR block
+type IPAddrDTO struct {
+	Start   string `json:"start,omitempty"`
+	End     string `json:"end,omitempty"`
+	CIDR    string `json:"cidr,omitempty"`
+	Address string `json:"address,omitempty"`
+}
+// AccountLevelIPDirectionalGroupDTO wraps an account-level, IP directional-group response
+type AccountLevelIPDirectionalGroupDTO struct {
+	Name        string      `json:"name"`
+	Description string      `json:"description"`
+	IPs         []IPAddrDTO `json:"ips"`
+}
+// AccountLevelGeoDirectionalGroupListDTO wraps a list of account-level, geo directional-groups response from a index request
+type AccountLevelGeoDirectionalGroupListDTO struct {
+	AccountName string                               `json:"zoneName"`
+	GeoGroups   []AccountLevelGeoDirectionalGroupDTO `json:"geoGroups"`
+	Queryinfo   QueryInfo                            `json:"queryInfo"`
+	Resultinfo  ResultInfo                           `json:"resultInfo"`
+}
+
+// AccountLevelIPDirectionalGroupListDTO wraps an account-level, IP directional-group response
+type AccountLevelIPDirectionalGroupListDTO struct {
+	AccountName string                              `json:"zoneName"`
+	IPGroups    []AccountLevelIPDirectionalGroupDTO `json:"ipGroups"`
+	Queryinfo   QueryInfo                           `json:"queryInfo"`
+	Resultinfo  ResultInfo                          `json:"resultInfo"`
+}
+
+// DirectionalPoolPath generates the URI for directional pools by account, type & slug ID
+func DirectionalPoolPath(acct, typ, slugID string) string {
```

Comment:
Do we need all of these url-helpers to be public? Are they being called from the terraform provider? Should they be?